integration-test/background/storage.js

```diff
@@ -19,73 +17,111 @@ describe(`On https://${testPageDomain}/privacy-protections/storage-blocking/`, (
-        const jsCookie = cookies.find(({ name, domain }) => name === 'jsdata' && domain === thirdPartyDomain)
-        expect(jsCookie).toBeTruthy()
-        expect(jsCookie.expires).toBeGreaterThan(Date.now() / 1000)
-    })
+        it('does not block 3rd party JS cookies not on block list', () => {
+            const jsCookie = cookies.find(({ name, domain }) => name === 'jsdata' && domain === thirdPartyDomain)
+            expect(jsCookie).toBeTruthy()
+            expect(jsCookie.expires).toBeGreaterThan(Date.now() / 1000)
+        })
+
+        it('blocks 3rd party JS cookies from trackers', () => {
+            const headerCookie = cookies.find(({ name, domain }) => name === 'jsdata' && domain === thirdPartyTracker)
+            expect(headerCookie).toBeUndefined()
+        })
 
-    it('does not block 1st party JS cookies set by non-trackers', () => {
-        const jsCookie = cookies.find(({ name, domain }) => name === 'tpsdata' && domain === testPageDomain)
-        expect(jsCookie).toBeTruthy()
-        expect(jsCookie.expires).toBeGreaterThan(Date.now() / 1000 + 950400) // 11 days in the future
+        it('does not block 1st party JS cookies set by non-trackers', () => {
+            const jsCookie = cookies.find(({ name, domain }) => name === 'tpsdata' && domain === testPageDomain)
+            expect(jsCookie).toBeTruthy()
+            expect(jsCookie.expires).toBeGreaterThan(Date.now() / 1000 + 950400) // 11 days in the future
+        })
+
+        it('reduces the expiry of 1st party JS cookies set by trackers to 8 days', () => {
+            const jsCookie = cookies.find(({ name, domain }) => name === 'tptdata' && domain === testPageDomain)
+            expect(jsCookie).toBeTruthy()
+            expect(jsCookie.expires).toBeGreaterThan(Date.now() / 1000)
+            expect(jsCookie.expires).toBeLessThan(Date.now() / 1000 + 864000) // 10 days in the future
+        })
     })
 
-    it('reduces the expiry of 1st party JS cookies set by trackers to 8 days', () => {
-        const jsCookie = cookies.find(({ name, domain }) => name === 'tptdata' && domain === testPageDomain)
-        expect(jsCookie).toBeTruthy()
-        expect(jsCookie.expires).toBeGreaterThan(Date.now() / 1000)
-        expect(jsCookie.expires).toBeLessThan(Date.now() / 1000 + 864000) // 10 days in the future
+    describe(`On https://${thirdPartyTracker}/privacy-protections/storage-blocking/`, () => {
+        it('does not block iFrame tracker cookies from same entity', async () => {
+            const { browser, page } = await setup()
+            async function waitForAllResults () {
+                while ((await page.$$('#tests-details > li > span > ul')).length < 2) {
+                    await new Promise(resolve => setTimeout(resolve, 100))
+                }
+            }
+            await page.goto(`https://${thirdPartyTracker}/privacy-protections/storage-blocking/?store`, { waitUntil: 'networkidle0' })
```

Comment:
Why aren't we using testPageDomain here? Isn't this changing outcome of test that we are using `broken.third-party.site` here?

Reply:
That's the purpose of this test - we're loading a tracker page and checking that the cookies for an iframe on the same origin (i.e. entity) are not blocked. We're testing that blocking works differently to the other block of tests where the iframe is 3rd party.